### PR TITLE
chore: [IOBP-933] Add exponential backoff retry into wallet section screen

### PR DESCRIPTION
**Pull request by Hantex9**
## Short description
This PR introduces the management of exponential backoff retry with a banner on the main "Wallet" screen. The banner will appear if at least one of the three requests from the B&P domain (idpay, cgn, and payment methods) fails. When one of the requests fails, tapping the CTA to retry will trigger all the failed requests. The retry logic follows an exponential backoff, where the wait time increases exponentially with each retry attempt to a maximum of 3 minutes.

## List of changes proposed in this pull request
- Added a `WalletCardsCategoryRetryErrorBanner` component that handles all the logic when it should be rendered (only when at least one of the three requests has failed)
- Managed the 404 of the CGN card in a different way, if we receive 404 from the backend it shouldn't be tracked as an error but as a none value (it means that the user doesn't have a CGN);

## How to test
- With the app in dev env and the io-dev-server started, try to hardcode a 500 error response on the dev-server in one of the three requests that are emitted inside the Wallet home screen (CGN, Payments methods and IDPay) [Reminder: If you want test IDPay cards, please enable the feature flag `IDPay` inside the settings screen;
- If the banner is visible, double-check that it also has the exponential backoff retry

## Preview
In this Preview, I forced the CGN API to respond 500 error

https://github.com/user-attachments/assets/d8405b94-671d-4270-987c-4f3f35e6f21b

### CODE REVIEW

File: ts/features/newWallet/components/WalletCardsCategoryRetryErrorBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      if (isPaymentMethodsError) {
+        dispatch(getPaymentsWalletUserMethods.request());
+      }
+      if (isCgnError) {
+        dispatch(cgnDetails.request());
+      }
+    }
+  };
+
+  if (isPaymentMethodsError || isCgnError || isIdPayError) {
```
Comment (LeleDallas) [inline ts/features/newWallet/components/WalletCardsCategoryRetryErrorBanner.tsx:49]: We can simplify this condition with `&&` operator
```tsx
return (
    (isPaymentMethodsError || isCgnError || isIdPayError) && (
      <BannerErrorState
        icon="warningFilled"
        label={I18n.t("features.wallet.home.otherMethods.error.banner.label")}
        actionText={I18n.t(
          "features.wallet.home.otherMethods.error.banner.cta"
        )}
        onPress={handleOnRetry}
      />
    )
```

File: ts/features/newWallet/components/WalletCardsCategoryContainer.tsx [hunk 2]
```diff
@@ -37,21 +38,34 @@ export const WalletCardsCategoryContainer = ({
...
+  ) : undefined;
+
+  return (
+    <Animated.View testID={testID} layout={LinearTransition.duration(200)}>
+      {header && <ListItemHeader {...header} />}
+      {React.isValidElement(topElement) && React.cloneElement(topElement)}
+      <Animated.FlatList
+        scrollEnabled={false}
+        data={cards}
+        ItemSeparatorComponent={() =>
```
Comment (LeleDallas) [inline ts/features/newWallet/components/WalletCardsCategoryContainer.tsx:57]: We can use the `&&` operator to make it more readable 
```ts
        ItemSeparatorComponent={() => !isStacked && <VSpacer size={16} />}
```

File: ts/features/newWallet/components/WalletCardsCategoryContainer.tsx [hunk 3]
```diff
@@ -37,21 +38,34 @@ export const WalletCardsCategoryContainer = ({
...
-      itemLayoutAnimation={itemLayoutAnimation}
-      entering={FadeInDown.duration(150)}
-      exiting={FadeOutDown.duration(150)}
-    />
-    <VSpacer size={24} />
-  </Animated.View>
-);
+}: WalletCardsCategoryContainerProps) => {
+  // Show the footer with the banner (if possible) to retry only if the category is of any domain of B&P (cgn, bonus or payment)
+  const ListFooter = cards.find(card => card.category !== "itw") ? (
```
Comment (LeleDallas) [inline ts/features/newWallet/components/WalletCardsCategoryContainer.tsx:43]: We can simplify with `&&` operator
```ts
const ListFooter = cards.find(card => card.category !== "itw") && (
    <>
      <VSpacer size={16} />
      <WalletCardsCategoryRetryErrorBanner />
    </>
  );

File: ts/features/newWallet/components/WalletCardsCategoryRetryErrorBanner.tsx (hunk 1 above)
Comment (Hantex9) [reply ts/features/newWallet/components/WalletCardsCategoryRetryErrorBanner.tsx]: Addressed into fe03880bc6b11c39ece9c43c35c7b94ace456a8e

File: ts/features/newWallet/components/WalletCardsCategoryContainer.tsx (hunk 2 above)
Comment (Hantex9) [reply ts/features/newWallet/components/WalletCardsCategoryContainer.tsx]: Addressed into fe03880bc6b11c39ece9c43c35c7b94ace456a8e

File: ts/features/newWallet/components/WalletCardsCategoryContainer.tsx (hunk 3 above)
Comment (Hantex9) [reply ts/features/newWallet/components/WalletCardsCategoryContainer.tsx]: Addressed into fe03880bc6b11c39ece9c43c35c7b94ace456a8e

**Review by LeleDallas — APPROVED** [review]

**Pull request merged**